### PR TITLE
Convert CreateCustomVariableEntityTest to MFTF

### PR DESCRIPTION
**Pull request by eduard13**
### Description (*)
This PR converts the CreateCustomVariableEntityTest to MFTF and covers the following scenario:

- Login into Admin Panel as admin
- Navigate to New Custom Variable page
- Save a new Custom Variable
- Checks if the Success Message is visible and if the new custom variable is visible in the grid

### Fixed Issues (if relevant)

1. #303 - Convert CreateCustomVariableEntityTest to MFTF

### CODE REVIEW

File: app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminAssertSuccessMessageActionGroup" >
```
Comment (rogyar) [inline app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml]: Please, use the following action group for this purpose:
https://github.com/magento/magento-functional-tests-migration/blob/2.3-develop/app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertMessageOnAdminLoginActionGroup.xml

File: app/code/Magento/Variable/Test/Mftf/Section/AdminVariableMessagesSection.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (rogyar) [inline app/code/Magento/Variable/Test/Mftf/Section/AdminVariableMessagesSection.xml]: Please, check my previous comment. Most likely, we don't need an additional messages section

File: dev/tests/functional/tests/app/Magento/Variable/Test/TestCase/CreateCustomVariableEntityTest.xml [hunk 3]
```diff
@@ -13,6 +13,7 @@
             <data name="customVariable/data/name" xsi:type="string">variableName%isolation%</data>
             <data name="customVariable/data/html_value" xsi:type="string">&lt;h1&gt;variableName%isolation%&lt;/h1&gt;</data>
             <data name="customVariable/data/plain_value" xsi:type="string">&lt;p&gt;variablePlainText%isolation%&lt;/p&gt;</data>
+            <data name="tag" xsi:type="string">mftf_migrated:yes</data>
```
Comment (rogyar) [inline dev/tests/functional/tests/app/Magento/Variable/Test/TestCase/CreateCustomVariableEntityTest.xml]: This section already contains the data named as "tag". Please, use the comma-separated approach as it's used here https://github.com/magento/magento-functional-tests-migration/pull/699/files#diff-7e8da2c64114791ac355dc4b3df3848aR11

**Comment by rogyar** [comment]
Hi @eduard13. Thank you for your contribution. Please, check my comments in the review. 
Also, please, consider prefixing action groups as Admin|Storefront|Assert. Also, if possible, split the action groups so we have one action group per file. If you need any help, do not hesitate to contact me in Slack.
Thank you.

File: app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml (hunk 1 above)
Comment (eduard13) [reply app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml]: The mentioned action group by you, is more related to login form (if we check the selector). So I thought that it's better to introduce a new action group in the Backend module, which can be used for any other tests.

File: app/code/Magento/Variable/Test/Mftf/Section/AdminVariableMessagesSection.xml (hunk 2 above)
Comment (eduard13) [reply app/code/Magento/Variable/Test/Mftf/Section/AdminVariableMessagesSection.xml]: Agree here (it wasn't on the moment when the test was written), I removed this one and reused the existing one.

File: dev/tests/functional/tests/app/Magento/Variable/Test/TestCase/CreateCustomVariableEntityTest.xml (hunk 3 above)
Comment (eduard13) [reply dev/tests/functional/tests/app/Magento/Variable/Test/TestCase/CreateCustomVariableEntityTest.xml]: Updated this as well.

**Comment by eduard13** [comment]
Hi @rogyar the mentioned notices were adjusted, moreover I've slightly adjusted the test and action groups, by moving some general Action groups (like filterGrid, resetFilter) into Backend module, which can be later used.
Could you please check these changes?

Thank you.

File: app/code/Magento/Backend/Test/Mftf/ActionGroup/AdminResetFilterActionGroup.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminResetFilterActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Backend/Test/Mftf/ActionGroup/AdminResetFilterActionGroup.xml]: Please rename as `AdminResetGridFilterActionGroup`

File: app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertValueInGridActionGroup.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertValueInGridActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertValueInGridActionGroup.xml]: `AssertValueInAdminGridActionGroup`

File: app/code/Magento/Backend/Test/Mftf/Section/AdminGridTableSection.xml [hunk 6]
```diff
@@ -10,5 +10,6 @@
           xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
     <section name="AdminGridTableSection">
         <element name="row" type="text" selector="table.data-grid tbody tr[data-role=row]:nth-of-type({{row}})" parameterized="true"/>
+        <element name="rowByValue" type="text" selector="//tr[@data-role='row'] //td[contains(., '{{var}}')]" parameterized="true" timeout="30"/>
```
Comment (rogyar) [inline app/code/Magento/Backend/Test/Mftf/Section/AdminGridTableSection.xml]: Please, clarify the selector by prefixing by `table.data-grid tbody`

File: app/code/Magento/Variable/Test/Mftf/ActionGroup/NavigateToNewVariablePageActionGroup.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="NavigateToNewVariablePageActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Variable/Test/Mftf/ActionGroup/NavigateToNewVariablePageActionGroup.xml]: Please, add `Admin` prefix

File: app/code/Magento/Variable/Test/Mftf/ActionGroup/NavigateToVariablesPageActionGroup.xml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="NavigateToVariablesPage">
```
Comment (rogyar) [inline app/code/Magento/Variable/Test/Mftf/ActionGroup/NavigateToVariablesPageActionGroup.xml]: Please, add Admin prefix and make sure the file name is the same as action group name

**Comment by eduard13** [comment]
Hi @rogyar the mentioned review notes has been adjusted, could you have a look please?
Thank you.

**Comment by eduard13** [comment]
Hi @okolesnyk, could you please check the last commit where I'm trying to create a new operation (create), but for some reasons it is failing?
Is there any way to debug such requests, to see what's happening?
Thank you.

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento-functional-tests-migration?pullRequest=700) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento-functional-tests-migration?pullRequest=700) before we can accept your contribution.<br/>

**Comment by eduard13** [comment]
@okolesnyk looks like it was removed after refactoring it. I've reverted it in order to keep the Backward Compatibility.
Thank you.

File: app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml (hunk 1 above)
Comment (rogyar) [reply app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml]: Hi @eduard13. Looks like we have the corresponding group currently in the mainline. Please, take a look at https://github.com/magento/magento-functional-tests-migration/blob/2.3-develop/app/code/Magento/Backend/Test/Mftf/ActionGroup/AssertMessageInAdminPanelActionGroup.xml

Comment (eduard13) [reply app/code/Magento/Variable/Test/Mftf/ActionGroup/AdminAssertSuccessAddedActionGroup.xml]: Hi @rogyar I've merged the mainline, which replaced this ActionGroup.

**Pull request merged**